File: src/main/java/com/mateusz/spring_boot_todos/repositories/UserRepository.java
package com.mateusz.spring_boot_todos.repositories;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.mateusz.spring_boot_todos.models.User;

public interface UserRepository extends JpaRepository<User, String> {
    Optional<User> findByUsername(String username);

}

File: src/main/kotlin/com/mateusz/todos/kotlin_sboot_todos_backend/repositories/UserRepository.kt
package com.mateusz.todos.kotlin_sboot_todos_backend.repositories

import com.mateusz.todos.kotlin_sboot_todos_backend.models.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User?, String?> {
    fun findByUsername(username: String?): Optional<User?>?
}